### PR TITLE
Replace grafana-agent with alloy

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/roadmap/issues/3521

Replace Grafana Agent with Alloy to import PrometheusRules into Mimir.

In this case Alloy is a drop-in replacement for Grafana Agent. So config stays exactly the same.

### CODE REVIEW

File: helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: Why did you remove this policy configuration ?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: Because the networkpolicy is directly within the helm chart of Alloy.

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: We need to push it  upstream then because @marieroque will not like this :D

File: helm/prometheus-rules/templates/alloy-rules.yaml
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alloy-rules.yaml:12]: Can you move alloy out of playground?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alloy-rules.yaml:12]: Which catalog do we want it into ?

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alloy-rules.yaml:12]: Either giantswarm or default I think

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/alloy-rules.yaml:12]: done

Comment (QuentinBisson) [reply helm/prometheus-rules/templates/alloy-rules.yaml:12]: Thanks :)

**Comment by QuentinBisson** [comment]
I'm fine with this but i'll let @marieroque take a look

File: helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml
Comment (marieroque) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: Maybe we need to add the CNP:
https://github.com/giantswarm/grafana-agent-app/blob/main/helm/grafana-agent/templates/cilium-networkpolicy.yaml
as manifest in that repo ?

Comment (marieroque) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: @TheoBrigitte did your tests work in term of communication ?
Did Hubble UI show no dropped request ?

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: I did not check Hubble UI, my test was successful (I deleted all the rules from Mimir, and Alloy re-imported them all)
Also there is a CNP already added in Alloy chart > https://github.com/giantswarm/alloy-app/blob/main/helm/alloy/templates/cilium-networkpolicy.yaml

Comment (TheoBrigitte) [reply helm/prometheus-rules/templates/grafana-agent-rules-configmap.yaml:6]: Just checked and there's no error in Hubble UI :heavy_check_mark:

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuentinBisson** [comment]
Before merging, how about we rename it to something that does not include the project name in it

**Review by marieroque — APPROVED** [review]

**Comment by TheoBrigitte** [comment]
> Before merging, how about we rename it to something that does not include the project name in it like mimir-rules-loader or smth? That way if we change to something else in thé future the component name does not change?

There might be a change that we will use this Alloy to also load rules into Loki, so I will keep this name for now and we can always rename this later on if there's a need for it.

**Comment by QuentinBisson** [comment]
Sure sure and we can rename later. I would like if we do not do the same mistake with the bundle later on when we introduce alloy 😅

**Comment by QuentinBisson** [comment]
I would hope that you remember this component is alloy marie 😅

**Pull request merged**
